rotkehlchen/data_import/importers/blockpit.py

```diff
@@ -0,0 +1,266 @@
+import csv
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+from uuid import uuid4
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.converters import LOCATION_TO_ASSET_MAPPING
+from rotkehlchen.assets.utils import symbol_to_asset_or_token
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.assets import A_USD
+from rotkehlchen.data_import.importers.constants import BLOCKPIT_EVENT_PREFIX
+from rotkehlchen.data_import.utils import (
+    BaseExchangeImporter,
+    UnsupportedCSVEntry,
+    detect_duplicate_event,
+)
+from rotkehlchen.db.drivers.gevent import DBCursor
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.exchanges.data_structures import AssetMovement, Trade
+from rotkehlchen.history.events.structures.base import HistoryEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_fee,
+    deserialize_timestamp_from_date,
+)
+from rotkehlchen.types import AssetMovementCategory, Fee, Location, Price, TradeType
+from rotkehlchen.utils.misc import ts_sec_to_ms
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import AssetWithOracles
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def exchange_row_to_location(entry: str) -> Location:
+    """Takes the Source Name from blockpit and returns a location"""
+    if entry == 'Binance':
+        return Location.BINANCE
+    if entry == 'Binance US':
+        return Location.BINANCEUS
+    if entry == 'Bitcoin.de':
```

Comment:
have you tried using Location.deserialize(entry). I believe it works for most of the cases care but probably you will need to handle the deserialization error to return the `external` location and the case of bitcoin.de I'm not sure if it will work